### PR TITLE
feat: Library

### PR DESCRIPTION
**Pull request by lafette21**

### CODE REVIEW

File: example/main.cc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    fmt::print("Example {}\n", "project");
+
+    try {
+        yui::gui gui{"Yui", yui::gui_config{ yui::vec2{ 1280, 720 }, {}, {} }};
+
+        gui.run([&gui] {
+            gui.main_menu().menu("Foo");
+        });
+    } catch (const std::exception& ex) {
+        fmt::print("Exception occured: {}", ex.what());
```
Comment (GK72) [inline example/main.cc]: Missing newline from print.

File: cmake/dependencies.cmake [hunk 2]
```diff
@@ -0,0 +1,4 @@
+find_package(fmt REQUIRED)
+find_package(glfw3 REQUIRED)
+find_package(GTest REQUIRED)
```
Comment (GK72) [inline cmake/dependencies.cmake]: Some libraries includes the necessary dependencies for testing only when testing is needed. Might be better in the `unit-test`'s cmake file.

Comment (lafette21) [reply cmake/dependencies.cmake]: Done.

File: example/main.cc (hunk 1 above)
Comment (lafette21) [reply example/main.cc]: Done.

**Review by GK72 — APPROVED** [review]
+3

**Pull request merged**